service/history/replication/task_processor_test.go

```diff
@@ -293,3 +300,46 @@ func (s *taskProcessorSuite) TestGenerateDLQRequest_ReplicationTaskTypeSyncActiv
 	s.Equal(runID, request.TaskInfo.GetRunID())
 	s.Equal(persistence.ReplicationTaskTypeSyncActivity, request.TaskInfo.GetTaskType())
 }
+
+func (s *taskProcessorSuite) TestTriggerDataInconsistencyScan_Success() {
+	domainID := uuid.New()
+	workflowID := uuid.New()
+	runID := uuid.New()
+	task := &types.ReplicationTask{
+		TaskType: types.ReplicationTaskTypeSyncActivity.Ptr(),
+		SyncActivityTaskAttributes: &types.SyncActivityTaskAttributes{
+			DomainID:    domainID,
+			WorkflowID:  workflowID,
+			RunID:       runID,
+			ScheduledID: 1,
+			Version:     100,
+		},
+	}
+	fixExecution := entity.Execution{
+		DomainID:   domainID,
+		WorkflowID: workflowID,
+		RunID:      runID,
+		State:      persistence.WorkflowStateCorrupted,
+	}
+	jsArray, err := json.Marshal(fixExecution)
+	s.NoError(err)
+	requestMatcher := ctesting.NewMatcher(func(x interface{}) bool {
```

Comment:
nit: I sometimes use `DoAndReturn` to validate the inputs. 
https://github.com/uber/cadence/blob/58e8e1ccb65b95f2a99ed4940a52336e15ab2aa4/service/history/task/cross_cluster_task_processor_test.go#L149

Reply:
Good to know.